From 17780bbb73835b31b44a00221828e7adec04fb8b Mon Sep 17 00:00:00 2001
From: Yue Ma <yuem@codeaurora.org>
Date: Mon, 17 Nov 2014 16:16:16 -0800
Subject: [PATCH] cnss: Add NULL check for PM related APIs

Since CMSS PM related APIs can be called at very early stage of WLAN
driver loading, add NULL check for them to avoid crashes if PCIe
enumeration fails which makes the penv variable as NULL.

Change-Id: I7c5e67cb1583b48b24ec04df21e482f4885d2e42
CRs-fixed: 758267
Signed-off-by: Yue Ma <yuem@codeaurora.org>
---
 drivers/net/wireless/cnss/cnss.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/net/wireless/cnss/cnss.c b/drivers/net/wireless/cnss/cnss.c
index f73e64d..c931722 100644
--- a/drivers/net/wireless/cnss/cnss.c
+++ b/drivers/net/wireless/cnss/cnss.c
@@ -2518,12 +2518,22 @@ static void __exit cnss_exit(void)
 
 void cnss_request_pm_qos(u32 qos_val)
 {
+	if (!penv) {
+		pr_err("%s: penv is NULL!\n", __func__);
+		return;
+	}
+
 	pm_qos_add_request(&penv->qos_request, PM_QOS_CPU_DMA_LATENCY, qos_val);
 }
 EXPORT_SYMBOL(cnss_request_pm_qos);
 
 void cnss_remove_pm_qos(void)
 {
+	if (!penv) {
+		pr_err("%s: penv is NULL!\n", __func__);
+		return;
+	}
+
 	pm_qos_remove_request(&penv->qos_request);
 }
 EXPORT_SYMBOL(cnss_remove_pm_qos);
-- 
1.8.2.1

